java-checks/src/main/java/org/sonar/java/checks/CheckList.java

```diff
@@ -217,634 +218,716 @@
+    // Rule classes are listed alphabetically
+    AssertionArgumentOrderCheck.class,
+    AssertionCompareToSelfCheck.class,
+    AssertionFailInCatchBlockCheck.class,
+    AssertionInThreadRunCheck.class,
+    AssertionInTryCatchCheck.class,
+    AssertionsCompletenessCheck.class,
+    AssertionsInTestsCheck.class,
+    AssertionTypesCheck.class,
+    AssertionsWithoutMessageCheck.class,
+    AssertJApplyConfigurationCheck.class,
+    AssertJAssertionsInConsumerCheck.class,
+    AssertJChainSimplificationCheck.class,
+    AssertJConsecutiveAssertionCheck.class,
+    AssertJContextBeforeAssertionCheck.class,
+    AssertJTestForEmptinessCheck.class,
+    AssertThatThrownByAloneCheck.class,
+    AssertTrueInsteadOfDedicatedAssertCheck.class,
+    BadTestClassNameCheck.class,
+    BadTestMethodNameCheck.class,
+    BooleanOrNullLiteralInAssertionsCheck.class,
+    CallSuperInTestCaseCheck.class,
+    ExpectedExceptionCheck.class,
+    IgnoredTestsCheck.class,
+    JUnitCompatibleAnnotationsCheck.class,
+    JUnit4AnnotationsCheck.class,
+    JUnit45MethodAnnotationCheck.class,
+    JUnit5DefaultPackageClassAndMethodCheck.class,
+    JUnit5SilentlyIgnoreClassAndMethodCheck.class,
+    JunitNestedAnnotationCheck.class,
+    MockingAllMethodsCheck.class,
+    MockitoAnnotatedObjectsShouldBeInitializedCheck.class,
+    MockitoArgumentMatchersUsedOnAllParametersCheck.class,
+    MockitoEqSimplificationCheck.class,
+    NoTestInTestClassCheck.class,
+    OneExpectedCheckedExceptionCheck.class,
+    OneExpectedRuntimeExceptionCheck.class,
+    RandomizedTestDataCheck.class,
+    SpringAssertionsSimplificationCheck.class,
+    ParameterizedTestCheck.class,
+    TestAnnotationWithExpectedExceptionCheck.class,
+    TestsStabilityCheck.class,
+    ThreadSleepInTestsCheck.class,
+    TooManyAssertionsCheck.class,
+    UnusedTestRuleCheck.class);
+
+  private static final List<Class<?>> ALL_CHECKS = Stream.of(JAVA_MAIN_CHECKS, JAVA_TEST_CHECKS)
+    .flatMap(List::stream).collect(Collectors.toList());
+
+  private static final Set<Class<? extends JavaCheck>> JAVA_CHECKS_NOT_WORKING_FOR_AUTOSCAN = Set.of(
+    // Symbolic executions rules
+    AllowXMLInclusionCheck.class,
+    BooleanGratuitousExpressionsCheck.class,
+    ConditionalUnreachableCodeCheck.class,
+    CustomUnclosedResourcesCheck.class,
+    DenialOfServiceXMLCheck.class,
+    DivisionByZeroCheck.class,
+    InvariantReturnCheck.class,
+    LocksNotUnlockedCheck.class,
+    MapComputeIfAbsentOrPresentCheck.class,
+    MinMaxRangeCheck.class,
+    NonNullSetToNullCheck.class,
+    NoWayOutLoopCheck.class,
+    NullDereferenceCheck.class,
+    ObjectOutputStreamCheck.class,
+    OptionalGetBeforeIsPresentCheck.class,
+    ParameterNullnessCheck.class,
+    RedundantAssignmentsCheck.class,
+    StreamConsumedCheck.class,
+    StreamNotConsumedCheck.class,
+    UnclosedResourcesCheck.class,
+    XmlParserLoadsExternalSchemasCheck.class,
+    XxeProcessingCheck.class,
```

Comment:
We should not hardcode the list of rules relying on SE, but dynamically exclude them, to be sure that we are not missing any.